spec/policy/url_rewriting/url_rewriting_spec.lua

```diff
@@ -83,54 +94,45 @@ describe('URL rewriting policy', function()
     end)
 
     it('can apply the "push" operation to args in the query', function()
-      stub(ngx.req, 'get_uri_args', function()
-        return { an_arg = '1'}
-      end)
-
       local config_to_add_args = {
         query_args_commands = {
-          { op = 'push', arg = 'an_arg', value = '2' }
+          { op = 'push', arg = 'an_arg', value = '1' }
         }
       }
       local url_rewriting = URLRewriting.new(config_to_add_args)
 
       url_rewriting:rewrite()
 
-      assert.stub(ngx.req.set_uri_args).was_called_with({ an_arg = { '1', '2' } })
+      assert.spy(spy_mocked_query_params_push).was_called_with(
```

Comment:
Looks like all these tests are sharing one spy. Is that safe?

Reply:
Changed. They need to be instantiated in the `before_each` to reset their calls.